server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -74,6 +77,61 @@ public class DecommissionController {
         this.threadPool = threadPool;
     }
 
+    /**
+     * This method sends a transport call to retry decommission action, given that -
+     * 1. cluster_manager_node_timeout is not timed out
+     * 2. And executed when there was a cluster manager change
+     *
+     * @param decommissionRequest decommission request object
+     * @param startTime start time of previous request
+     * @param listener callback for the retry action
+     */
+    public void retryDecommissionAction(
+        DecommissionRequest decommissionRequest,
+        long startTime,
+        ActionListener<DecommissionResponse> listener
+    ) {
+        final long remainingTimeoutMS = decommissionRequest.timeout().millis() - (threadPool.relativeTimeInMillis() - startTime);
+        if (remainingTimeoutMS <= 0) {
+            String errorMsg = "cluster manager node timed out before retrying ["
+                + DecommissionAction.NAME
+                + "] for attribute ["
+                + decommissionRequest.getDecommissionAttribute()
+                + "] after cluster manager change";
+            logger.debug(errorMsg);
+            listener.onFailure(new OpenSearchTimeoutException(errorMsg));
+            return;
+        }
+        decommissionRequest.setRetryOnClusterManagerChange(true);
+        decommissionRequest.setTimeout(TimeValue.timeValueMillis(remainingTimeoutMS));
+        transportService.sendRequest(
+            transportService.getLocalNode(),
```

Comment:
This will be routed internally since the `TransportDecommissionAction` extends `TransportClusterManagerNodeAction`.

Reply:
Can we avoid that hop and sent the request directly to the new cluster manager

Reply:
I am assuming the current transport request will gracefully finish and don't suffer from timeouts while we fire a fresh new retryable request on the new leader

Reply:
If the node went down altogether then would it even trigger the retry? Because ultimately responsibility of triggering a retry is with abdicated leader. Two scenarios here -
1. Send the transport request to local node which is also a abdicated leader. Here, when we send the request, it will form the local node connection with itself. And based on the action it routes it to active leader
2. Send the transport request to the newly elected leader. Here the abdicated leader will try to form connection with newly elected leader and then send request to it and then the newly elected leader based on the action will forward the request again (which in this case will be to itself)

Reply:
If we see it this way, number of hops is less in 1st scenario. Let me know of I am missing anything